mptcp: Fix add_addr hmac computation and compiler warnings

cpaasch opened the pull request:
I see that the ADD_ADDR-hmac is incorrectly calculated. The reason is because the key_a/b is an unsigned long and not a u64. Thus, when doing *key_a = key1, key1 will be truncated to 32bit.

Also, address some compile-time issues on my Ubuntu platform that I use for testing.

matttbe commented on gtests/net/packetdrill/mptcp.c:
```diff
@@ -1904,8 +1904,8 @@ static u64 add_addr_ipv4_hmac(u64 key1, u64 key2, u8 address_id, struct in_addr*
-	unsigned long *key_a = (unsigned long*)hmac_key;
-	unsigned long *key_b = (unsigned long*)&(hmac_key[8]);
+	u64 *key_a = (u64 *)hmac_key;
+	u64 *key_b = (u64 *)&(hmac_key[8]);
```
no need to do these modifications in `mp_join_syn_ack()` when the key for HMAC-SHA1 is built?

https://github.com/multipath-tcp/packetdrill/blob/24aba7b82fd620468c45be4f087dab803a5bfcc4/gtests/net/packetdrill/mptcp.c#L821-L826

cpaasch replied:
Good catch! Same there indeed.

matttbe commented on gtests/net/packetdrill/run_system_call.c:
```diff
@@ -28,6 +28,7 @@
 #include <arpa/inet.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <inttypes.h>
```
(thank you for the fixes here in this file but it is maybe clearer in a dedicated commit :-P )

cpaasch replied:
Sure, I can split out the changes.

matttbe reviewed (commented):
Thank you for the fix, good catch!

Just one question, the rest looks good to me!

cpaasch committed:
mptcp: Fix hmac computation

I see that the ADD_ADDR-hmac is incorrectly calculated. The reason is
because the key_a/b is an unsigned long and not a u64. Thus, when doing
*key_a = key1, key1 will be truncated to 32bit.

Same problem for the MP_JOIN SYN.

Signed-off-by: Christoph Paasch <cpaasch@apple.com>

cpaasch committed:
packetdrill: Fix compile-time errors

Had a few compile-time errors on my hosts. Fix those.

Signed-off-by: Christoph Paasch <cpaasch@apple.com>

matttbe reviewed (approved):
thank you for the new version, looks good to me!

I will wait for @dcaratti before merging this.

dcaratti commented:
LGTM, thanks!

matttbe commented:
I also merged this in `mptcp-6.0` branch as it is also helpful for -net
